source/core/Events.cpp

```diff
@@ -586,6 +650,60 @@ static inline InstructionCore *ReturnRegForEventsFatalError(const char *errorMes
     return THREAD_EXEC()->Stop();
 }
 
+// Recursive helper function to handle event registration (or re-registration) of polymorphic inputs (clusters/arrays of refnums)
+LVError RegisterForEventsCore(EventQueueID qID, DynamicEventRegInfo *regInfo, Int32 refInput, TypeRef refType, EventSource eSource,
```

Comment:
This function appears to be a duplicate of DynamicEventMatchCore. Is it possible to share code?

Reply:
I thought about this when I wrote them. Their recursive structure is similar, but they have very different arguments and do different things in the base case; I couldn't find a clean way to share code.